### ISSUE TITLE
location filter uses lowercase only and ignores locations with dashes

### ISSUE THREAD

**Issue by aleksandra-kim** [brightway-lca/brightway2-data#35]
**[Original report](https://bitbucket.org/cmutel/brightway2-data/issue/35) by Pascal Lesage (Bitbucket: [_MPa_](https://bitbucket.org/_MPa_), ).**

----------------------------------------

Two cases were found:  

1) Filtering on location using uppercase (as they are found in the activity) returns onthing (i.e. an empty list). Changing to lowercase works, even though in reality e.g. 'DE' != 'de'  

2) Filtering on locations with dashes (as they are found in the activities) returns nothing (i.e. an empty list). Changing to lowercase in this case doesn't change anything, e.g. 'ca-qc', CA-QC', 'caqc', CAQC' all return nothing.

**Comment by aleksandra-kim** [comment]
**Original comment by Chris Mutel (Bitbucket: [cmutel](https://bitbucket.org/cmutel), GitHub: [cmutel](https://github.com/cmutel)).**

----------------------------------------

`CA-QC` is a separate issue, as `-` is a [special character in the search engine](http://whoosh.readthedocs.org/en/latest/releases/2_0.html?highlight=hyphen). Fixing this will [be complicated](https://groups.google.com/forum/#!topic/whoosh/n5i5J8B8qt8), and is a bit too much off the path for Brightway2 development, at least for me. You can always do the following:
```
#!python

[x for x in Database("foo").search("bar", filter={'location': "ca"}) if x['location'] == 'QC-CA']

```

I will make a post to the Whoosh mailing list about the first case; note that the [default analyzer is case-insensitive](https://groups.google.com/forum/#!msg/whoosh/DYftL2Gnpr8/g72rfjIyE6kJ), but this is maybe applied inconsistently?

**Comment by aleksandra-kim** [comment]
**Original comment by Chris Mutel (Bitbucket: [cmutel](https://bitbucket.org/cmutel), GitHub: [cmutel](https://github.com/cmutel)).**

----------------------------------------

[Post to Whoosh list](https://groups.google.com/forum/#!topic/whoosh/98Iso58KKkA)

**Comment by aleksandra-kim** [comment]
**Original comment by Chris Mutel (Bitbucket: [cmutel](https://bitbucket.org/cmutel), GitHub: [cmutel](https://github.com/cmutel)).**

----------------------------------------

Reply from Matt Chaput (whoosh author):

    When you use the query parser, it runs the analyzer on the text and builds the query objects from the output.
    If you construct query objects directly, you are giving them the *exact* text to search for.

Our search filter uses a query object:

    And([Term(k, v) for k, v in filter.items()])

However, it is not yet clear to me how to work with the Whoosh machinery to make this case-insensitive.

**Comment by aleksandra-kim** [comment]
**Original comment by Bernhard Steubing (Bitbucket: [bsteubing](https://bitbucket.org/bsteubing), GitHub: [bsteubing](https://github.com/bsteubing)).**

----------------------------------------

also the "mask" option does only work with lowercase.
Knowing this limitation, couldn't you simply convert all user input to location, mask, etc. to lowercase? Just an idea.

**Comment by aleksandra-kim** [comment]
**Original comment by Chris Mutel (Bitbucket: [cmutel](https://bitbucket.org/cmutel), GitHub: [cmutel](https://github.com/cmutel)).**

----------------------------------------

All search data stored as lowercase. Fixes #35. To get this beahviour in an existing database, use 'db.make_searchable(reset=True'

**Comment by aleksandra-kim** [comment]
**Original comment by Chris Mutel (Bitbucket: [cmutel](https://bitbucket.org/cmutel), GitHub: [cmutel](https://github.com/cmutel)).**

----------------------------------------

Note that the fix doesn't help terms that have a dash, these have to be filtered outside the Whoosh search index.
